src/main/java/com/icthh/xm/ms/dashboard/repository/impl/ConfigDashboardRefreshableRepository.java

```diff
@@ -0,0 +1,132 @@
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.AntPathMatcher;
+
+@Transactional
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ConfigDashboardRefreshableRepository implements RefreshableConfiguration {
+
+    private static final String TENANT_NAME = "tenantName";
+
+    private final Map<String, Map<String, DashboardDto>> dashboardsByTenantByFile = new ConcurrentHashMap<>();
+
+    private final AntPathMatcher matcher = new AntPathMatcher();
+    private final ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+    private final ApplicationProperties applicationProperties;
+    private final TenantContextHolder tenantContextHolder;
+    private final IdRepository idRepository;
+    private final TenantConfigRepository tenantConfigRepository;
+
+    @Override
+    public void onRefresh(String updatedKey, String config) {
+        try {
+            String pathPattern = getPathPattern(updatedKey);
+            String tenant = matcher.extractUriTemplateVariables(pathPattern, updatedKey).get(TENANT_NAME);
+            updateByFileState(updatedKey, config, tenant);
+            log.info("Specification was for tenant {} updated from file {}", tenant, updatedKey);
+        } catch (Exception e) {
+            log.error("Error read xm specification from path " + updatedKey, e);
+        }
+    }
+
+
+    @SneakyThrows
+    private void updateByFileState(String updatedKey, String config, String tenant) {
+        Map<String, DashboardDto> dashboards = dashboardsByTenantByFile
+            .computeIfAbsent(tenant, key -> new LinkedHashMap<>());
+
+        if (StringUtils.isBlank(config)) {
+            dashboards.remove(updatedKey);
+            return;
+        }
+
+        DashboardDto dashboard = mapper.readValue(config, DashboardDto.class);
+        dashboards.put(updatedKey, dashboard);
+    }
+
+    @Override
+    public boolean isListeningConfiguration(String updatedKey) {
+        String tenantDashboardsFolderPathPattern = applicationProperties.getTenantDashboardsFolderPathPattern();
+        return matcher.match(tenantDashboardsFolderPathPattern, updatedKey);
+    }
+
+    @Override
+    public void onInit(String configKey, String configValue) {
+        onRefresh(configKey, configValue);
+    }
+
+    private String getPathPattern(String updatedKey) {
+        String tenantDashboardsFolderPathPattern = applicationProperties.getTenantDashboardsFolderPathPattern();
+        if (matcher.match(tenantDashboardsFolderPathPattern, updatedKey)) {
```

Comment:
duplicate check (was checked in isListeningConfiguration )

Reply:
done